airflow/models/connection.py

```diff
@@ -84,6 +84,7 @@ class Connection(Base, LoggingMixin):
         ('segment', 'Segment',),
         ('azure_data_lake', 'Azure Data Lake'),
         ('azure_container_instances', 'Azure Container Instances'),
+        ('azure_kubernetes_instances', 'Azure Kubernetes Instances'),
```

Comment:
Is this required? Google Cloud Platform has only one hook. GCP SQL has a separate, but this is a special case.
I am calling @asbh, because we deal with "creativity" of Azura in (another PR)[https://github.com/apache/airflow/pull/4524], and this change is loosely connected. We probably have to make systemic decisions on how to limit the creativity of Azure programmers.

Reply:
@apraovjr However, you would like to know that we are very grateful for this PR, but I think that we should reflect on certain matters before this PR can be accepted. In the last 3 days, this is second PR related to Azure that has similar comments.
https://github.com/apache/airflow/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aopen+Azure